Sources/Swift/Tools/ViewCapture/SentryViewPhotographer.swift

```diff
@@ -0,0 +1,76 @@
+#if canImport(UIKit) && !SENTRY_NO_UIKIT
+#if os(iOS) || os(tvOS)
+
+@_implementationOnly import _SentryPrivate
+import CoreGraphics
+import Foundation
+import UIKit
+
+@objcMembers
+class SentryViewPhotographer: NSObject, SentryViewScreenshotProvider {
+    private let redactBuilder: UIRedactBuilder
+    private let maskRenderer: SentryMaskRenderer
+    private let dispatchQueue = SentryDispatchQueueWrapper()
+
+    var renderer: SentryViewRenderer
+
+    init(
+        renderer: SentryViewRenderer,
+        redactOptions: SentryRedactOptions
+    ) {
+        self.renderer = renderer
+        self.maskRenderer = SentryDefaultMaskRenderer()
```

Comment:
The mask renderer uses Swift-only structs, so it can not be used from Objective-C.
Therefore the mask renderer can not be injected but is abstracted using the protocol `SentryMaskRenderer`